taskcluster/ac_taskgraph/loader/build_config.py

```diff
@@ -8,20 +8,33 @@
 
 from copy import deepcopy
 from taskgraph.loader.transform import loader as base_loader
+from taskgraph.util.templates import merge
 
 from ..build_config import get_components
 
 
 def loader(kind, path, config, params, loaded_tasks):
-    config['jobs'] = jobs = {
-        component['name']: {}
+    jobs = {
+        '{}{}'.format(
+            '' if build_type == 'regular' else build_type + '-',
+            component['name']
+        ): {
+            'attributes': {
+                'build-type': build_type,
+                'component': component['name'],
+            }
+        }
         for component in get_components()
+        for build_type in ('regular', 'release', 'snapshot')
+        if component['shouldPublish'] or build_type == 'regular'
     }
-    jobs.update(config.pop('overriden-jobs', {}))
+    jobs = merge(jobs, config.pop('overriden-jobs', {}))
 
     for job_name, job_config in config.pop('additional-jobs', {}).iteritems():
```

Comment:
I haven't looked deeply at any of the jobs, but I wonder if it would make more sense for `additional-jobs` (and maybe also `overriden-jobs`) to be in separate kinds, rather than in the build kind?  Or maybe the stuff in overriden jobs could be moved to `.build-config.yml`?

It looks like at leas the `additional-jobs` are different enough that they don't gain by being in the same kind.

Reply:
You're right. `additional-jobs` is better in its own kind. That allowed me to get rid of `default-gradle-commands` too! Thanks for pointing this out!